src/thunderbots/software/util/canvas_messenger/canvas_messenger.cpp

```diff
@@ -74,8 +89,11 @@ void CanvasMessenger::publishPayload(uint8_t layer, std::vector<Sprite> sprites)
     thunderbots_msgs::CanvasLayer new_layer;
     new_layer.data = payload;
 
-    // and publish
-    this->publisher.publish(new_layer);
+    // and publish if we have a valid ROS publisher
```

Comment:
Explain the scenario where we don't have a valid publisher (ie. unit tests without ROS running). Useful for context.

Reply:
Updated comment to:
```
    // and publish if we have a valid ROS publisher.
    // This check is important for cases where we're running this without running ROS,
    // such as in unit tests.
```